src/scene/graph-node.js

```diff
@@ -108,8 +108,17 @@ pc.extend(pc, function () {
 
         set: function (enabled) {
             if (this._enabled !== enabled) {
+                var prevValue = this._enabled;
+
                 this._enabled = enabled;
 
+                if (this.fire) {
+                    // TODO Using set_enabled here because it matches the convention in
+                    //      components/component.js, but we also have set:foo elsewhere.
+                    //      Which one is preferred?
+                    this.fire('set_enabled', 'enabled', prevValue, enabled);
```

Comment:
The reason I'm doing this is in order to remove this entity from the layout flow – i.e. if the user disables a child entity, we need to reflow without that entity taking up any space in the layout (and vice versa).

Reply:
Well it's just introducing new events to entities is a big deal in general at least in my mind, because then everyone is going to start using them for their own thing. When an entity gets disabled we fire onDisable an all its components. What if you handle that in the component that got disabled and fire an event there for the interested components to handle?

Reply:
Yeah I definitely appreciate it’s a big deal adding this to GraphNode. Cool – I’ll dispatch an event from onDisable/onEnable in Element instead.